### PR TITLE
Support events using protocol buffers

### PR DESCRIPTION
**Pull request by udaysagar2177**
Add example on how to use protobuf to send datapoints and events
Updated readme
Update maven compiler source to 1.7

### CODE REVIEW

File: signalfx-java-examples/pom.xml [hunk 1]
```diff
@@ -2,16 +2,18 @@
...
   <version>0.0.25-SNAPSHOT</version>
   <packaging>jar</packaging>
 
-  <name>signalfx.yammer.example</name>
+  <name>signalfx.example</name>
   <url>http://maven.apache.org</url>
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (tedoc2000) [inline signalfx-java-examples/pom.xml]: This should be 1.6 that is the earliest version we support

File: README.md [hunk 2]
```diff
@@ -373,5 +384,24 @@ try (AggregateMetricSender.Session i = mf.createSession()) {
                     .setKey("source")
                     .setValue("java"))
             .build());
+
+    i.setEvent(
+            SignalFxProtocolBuffers.Event.newBuilder()
+                  .setEventType("Deployments")
+                  .setCategory(SignalFxProtocolBuffers.EventCategory.ALERT)
```
Comment (jloomis) [inline README.md]: nit: EventCategory.ALERT doesn't make sense with eventType 'Deployments'

File: signalfx-java/src/main/java/com/signalfx/metrics/flush/AggregateMetricSender.java [hunk 3]
```diff
@@ -224,6 +245,23 @@ public void close() {
                     return;
                 }
             }
+
```
Comment (tedoc2000) [inline signalfx-java/src/main/java/com/signalfx/metrics/flush/AggregateMetricSender.java]: Remove extra whitespace

**Comment by udaysagar2177** [comment]
@mdubbyap @tedoc2000 
removed the sf_source update from this pull request and addressed above comments.

File: signalfx-java-examples/src/main/java/com/signalfx/example/ProtobufExample.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
+package com.signalfx.example;
```
Comment (mdubbyap) [inline signalfx-java-examples/src/main/java/com/signalfx/example/ProtobufExample.java]: New files should have our copyright at the top.

/**
 * Copyright (C) 2016 SignalFx, Inc. All rights reserved.
 */

**Pull request merged**